BaragonService/src/main/java/com/hubspot/baragon/service/managers/RequestManager.java

```diff
@@ -249,6 +254,12 @@ public BaragonResponse enqueueRequest(BaragonRequest request) throws RequestAlre
       throw new InvalidRequestActionException("The REVERT action may only be used internally by Baragon, you may specify UPDATE, DELETE, RELOAD, or leave the action blank(UPDATE)");
     }
 
+    if (request.isNoDuplicateUpstreams()) {
+      if (!validateNoDuplicateUpstreams(request.getAddUpstreams())) {
+        throw new InvalidUpstreamsException("If noDuplicateUpstreams is specified, you cannot have duplicate upstreams");
```

Comment:
Nit picky thing. Can we throw this instead inside validateNoDuplicateUpstreams and include as part of the message the actual upstreams that are duplicated? Would go a long way for debugging purposes to instantly know which host:port was in error.